de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/modelvalidator/C_OrderLine.java

```diff
@@ -59,4 +69,28 @@ public void deleteOlas(final I_C_OrderLine ol)
 			InterfaceWrapperHelper.delete(ola);
 		}
 	}
+
+	@ModelChange(timings = {ModelValidator.TYPE_BEFORE_CHANGE, ModelValidator.TYPE_BEFORE_NEW })
+	public void updateQtyAvailableToPromise(final I_C_OrderLine ol)
+	{
+		 final AvailableToPromiseRepository stockRepository = Adempiere.getBean(AvailableToPromiseRepository.class);
+		 final AvailableToPromiseQueryBuilder stockQueryBuilder = AvailableToPromiseQuery.builder()
+					.warehouseIds(ImmutableSet.of(ol.getM_Warehouse_ID()))
```

Comment:
C_OrderLine.M_Warehouse_ID is not mandatory..are you sure that it's always set?

Reply:
@metas-ts We have this callout de.metas.adempiere.callout.OrderLine.warehouse(ICalloutField).
Plus , the default value for AD_Column C_Orderline.M_Warewhouse_ID is @M_Warehouse_ID@
I think the warehouse is always set in order line.